src/pages/RFOX/components/Stake/StakeStatus.tsx

```diff
@@ -54,6 +58,12 @@ export const StakeStatus: React.FC<StakeRouteProps & StakeStatusProps> = ({
 
   const tx = useAppSelector(state => selectTxById(state, serializedTxIndex))
 
+  useEffect(() => {
+    if (tx?.status !== TxStatus.Confirmed) return
+
+    handleTxConfirmed()
+  }, [handleTxConfirmed, queryClient, tx?.status])
```

Comment:
Is there a reason why we have the queryClient as a dependency? And why don't we have it on `ChangeAddressStatus` side

Reply:
Nice catch - 557045eddb. The issue here isn't so much we have it as a dep but rather we shouldn't have `queryClient` in this component anymore altogether.